samplers/count_aware_sampler_test.go

```diff
@@ -0,0 +1,36 @@
+package samplers
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/thundra-io/thundra-lambda-agent-go/config"
+	"github.com/thundra-io/thundra-lambda-agent-go/constants"
+)
+
+func TestDefaultCountFreq(t *testing.T) {
+
+	cas := NewCountAwareSampler()
+
+	assert.Equal(t, int64(constants.DefaultSamplingCountFreq), cas.(*countAwareSampler).countFreq)
+}
+
+func TestCountFreqFromEnv(t *testing.T) {
+	config.SamplingCountFrequency = 10
+	cas := NewCountAwareSampler()
+
+	assert.Equal(t, int64(config.SamplingCountFrequency), cas.(*countAwareSampler).countFreq)
+}
+
+func TestFreqFromParam(t *testing.T) {
+	cas := NewCountAwareSampler(10)
+
+	assert.Equal(t, int64(10), cas.(*countAwareSampler).countFreq)
+}
+
+func TestSampledCountAware(t *testing.T) {
+	cas := NewCountAwareSampler(2)
+
+	assert.True(t, cas.IsSampled(nil))
+	assert.False(t, cas.IsSampled(nil))
```

Comment:
I think It would be better to add one more line to completely be sure that sampler is using the frequency value 2, like as the following:
```go
assert.True(t, cas.IsSampled(nil))
assert.False(t, cas.IsSampled(nil))
assert.True(t, cas.IsSampled(nil))
```
However, when I do that the test fails, because the sampler is using the frequency value 10 remained from the previous tests above (even though it is given 2 using a parameter). We can make sure that previous tests do not have side effects on this test, by clearing the `config.SamplingCountFrequency` value before starting the test.